extras/jslibfetcher.py

```diff
@@ -230,308 +234,162 @@ def get_pattern(prefix, url_pattern, library):
+
+
+def get_pattern(prefix, url_pattern, library):
+    for rule, versions in library.iteritems():
+        for version in versions:
+            url = url_pattern % version
+            process(url, rule, "%s.%s.%s" % (prefix, version, url.split("/")[-1]))
+
 
-get_pattern(
-    "crypto_js", "http://crypto-js.googlecode.com/svn/tags/3.0.2/src/%s.js",
-    CRYPTO_FILES)
+def get_patterns():
+    # AngularJS
+    get_pattern("angularjs",
+                "https://code.angularjs.org/%s/angular.js",
+                ANGULARJS_VERSIONS)
+    get_pattern("angularjs",
+                "https://code.angularjs.org/%s/angular.min.js",
+                ANGULARJS_VERSIONS)
+
+    # Backbone
+    get_pattern("backbone",
+                "https://raw.githubusercontent.com/jashkenas/backbone/%s/backbone.js",
+                BACKBONE_VERSIONS)
+    get_pattern("backbone",
+                "https://raw.githubusercontent.com/jashkenas/backbone/%s/backbone-min.js",
+                BACKBONE_VERSIONS)
+
+    # Dojo Toolkit
+    get_pattern("dojo",
+                "https://download.dojotoolkit.org/release-%s/dojo.js",
+                DOJO_VERSIONS)
+    get_pattern("dojo",
+                "https://download.dojotoolkit.org/release-%s/dojo.js.uncompressed.js",
+                DOJO_VERSIONS)
+
+    # jQuery
+    get_pattern("jquery",
+                "https://code.jquery.com/jquery-%s.js",
+                JQUERY_VERSIONS)
+    get_pattern("jquery",
+                "https://code.jquery.com/jquery-%s.min.js",
+                JQUERY_VERSIONS)
+
+    # jQueryUI
+    get_pattern("jqueryui",
+                "https://code.jquery.com/ui/%s/jquery-ui.min.js",
+                JQUERYUI_VERSIONS)
+    get_pattern("jqueryui",
+                "https://code.jquery.com/ui/%s/jquery-ui.js",
+                JQUERYUI_VERSIONS)
+
+    # moment.js
+    get_pattern("moment",
+                "https://raw.githubusercontent.com/moment/moment/%s/moment.js",
+                MOMENTJS_VERSIONS)
+    get_pattern("moment",
+                "https://raw.githubusercontent.com/moment/moment/%s/min/moment.min.js",
+                MOMENTJS_VERSIONS)
+
+    # MooTools
+    get_pattern("mootools",
+                "https://ajax.googleapis.com/ajax/libs/mootools/%s/mootools-yui-compressed.js",
+                MOOTOOLS_VERSIONS)
+    get_pattern("mootools",
+                "https://ajax.googleapis.com/ajax/libs/mootools/%s/mootools.js",
+                MOOTOOLS_VERSIONS)
+
+    # Prototype.js
+    get_pattern("prototype",
+                "https://ajax.googleapis.com/ajax/libs/prototype/%s/prototype.js",
+                PROTOTYPE_VERSIONS)
+
+    # Underscore
+    get_pattern("underscore",
+                "https://raw.github.com/documentcloud/underscore/%s/underscore.js",
+                UNDERSCORE_VERSIONS)
+    get_pattern("underscore",
+                "https://raw.github.com/documentcloud/underscore/%s/underscore-min.js",
+                UNDERSCORE_VERSIONS)
+
+    CRYPTO_FILES = {
```

Comment:
This has been like that before, I did not change the order. I think it would make more sense to keep this close to the corresponding `get_pattern` because it does contain versions of one single library, but instead uses the a particular (=the latest) version for several files of a library collection.